src/main/java/com/google/firebase/internal/FcmOptionsUtil.java

```diff
@@ -0,0 +1,15 @@
+package com.google.firebase.internal;
+
+import java.util.regex.Pattern;
+
+public final class FcmOptionsUtil {
+
+  private static final Pattern ANALYTICS_LABEL_REGEX = Pattern.compile("^[a-zA-Z0-9-_.~%]{0,50}$");
```

Comment:
Nit: Perhaps add a comment describing what the accepted pattern is.

Reply:
done.